src/dev_process.erl

```diff
@@ -191,23 +191,18 @@ do_compute(ProcID, Msg1, Msg2, TargetSlot, Opts) ->
                 ),
             case CurrentSlot rem Freq of
                 0 ->
-                    case snapshot(Msg3, Msg2, Opts) of
-                        {ok, Snapshot} ->
-                            ?event(snapshot,
-                                {got_snapshot, 
-                                    {storing_as_slot, CurrentSlot}
-                                }
-                            ),
-                            dev_process_cache:write(
-                                ProcID,
-                                CurrentSlot,
-                                Snapshot,
-                                Opts
-                            );
-                        not_found ->
-                            ?event(no_result_for_snapshot),
-                            nothing_to_store
-                    end;
+                    {ok, Snapshot} = snapshot(Msg3, Msg2, Opts),
+                    ?event(snapshot,
+                        {got_snapshot, 
+                            {storing_as_slot, CurrentSlot}
+                        }
+                    ),
+                    dev_process_cache:write(
+                        ProcID,
+                        CurrentSlot,
+                        Snapshot,
+                        Opts
+                    );
```

Comment:
This one is interesting. Are we certain that the `snapshot` function will never return not found? For example, if the `execution-device` does not provide snapshotting?

Reply:
My assumption was only the code itself on the function's implementation level.
From my understanding `snapshot` function always return 
```erlang
{ok,
        hb_private:set(
            hb_converge:set(
                SnapshotMsg,
                #{ <<"cache-control">> => [<<"store">>] },
                Opts
            ),
            #{ <<"priv/additional-hashpaths">> =>
                    [
                        hb_path:to_binary([ProcID, <<"Snapshot">>, Slot])
                    ]
            },
            Opts
        )
    }.
```